src/components/SiderMenu/BaseMenu.tsx

```diff
@@ -44,18 +45,21 @@ export default function BaseMenu (props: GlobalHeaderRightProps) {
    * 获得菜单子节点
    * @memberof SiderMenu
    */
-  function getNavMenuItems (menusData, parent?) {
+  function getNavMenuItems (menusData: Array<MenuDataProps>, parent?: string) {
     if (!menusData) {
       return []
     }
-    return menusData
-      .filter(item => item.name && !item.hideInMenu)
-      .map(item => getSubMenuOrItem(item, parent))
-      .filter(item => item)
+    let returnData = []
+    for (let item in menusData) {
+      if(menusData[item].name && menusData[item].hideInMenu){
+        getSubMenuOrItem(item, parent) && returnData.push(getSubMenuOrItem(item, parent))
```

Comment:
don't call  `getSubMenuOrItem` twice. Create a variable and then push it to the `returnData`

Reply:
done